core/services/ardupilot_manager/main.py

```diff
@@ -88,34 +77,29 @@ def update_endpoints(endpoints: Set[Endpoint] = Body(...)) -> Any:
     summary="Retrieve dictionary of available firmwares versions with their respective URL.",
 )
 @version(1, 0)
-def get_available_firmwares(response: Response, vehicle: Vehicle) -> Any:
-    try:
-        if not autopilot.current_board:
-            raise RuntimeError("Cannot fetch available firmwares as there's no board running.")
-        return autopilot.get_available_firmwares(vehicle, autopilot.current_board.platform)
-    except Exception as error:
-        response.status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
-        return {"message": f"{error}"}
+def get_available_firmwares(vehicle: Vehicle) -> Any:
+    if not autopilot.current_board:
+        raise RuntimeError("Cannot fetch available firmwares as there's no board running.")
+    return autopilot.get_available_firmwares(vehicle, autopilot.current_board.platform)
 
 
 @app.post("/install_firmware_from_url", summary="Install firmware for given URL.")
 @version(1, 0)
-async def install_firmware_from_url(response: Response, url: str) -> Any:
+async def install_firmware_from_url(url: str) -> Any:
     try:
         if not autopilot.current_board:
             raise RuntimeError("Cannot install firmware as there's no board running.")
         await autopilot.kill_ardupilot()
         autopilot.install_firmware_from_url(url, autopilot.current_board)
     except Exception as error:
-        response.status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
-        return {"message": f"{error}"}
+        raise error
```

Comment:
You don't need to raise error, you can remove the try block.

Reply:
We need the try/except/finally block to ensure we start the autopilot if everything goes wrong.